test/uk/gov/hmrc/helptosavefrontend/controllers/HelpToSaveAuthSpec.scala

```diff
@@ -16,55 +16,57 @@
+import play.api.test.Helpers.{contentAsString, redirectLocation}
 import uk.gov.hmrc.auth.core.AuthorisationException.fromString
-import uk.gov.hmrc.auth.core._
-import uk.gov.hmrc.helptosavefrontend.TestSupport
+import uk.gov.hmrc.helptosavefrontend.config.FrontendAppConfig
 import uk.gov.hmrc.helptosavefrontend.config.FrontendAppConfig.{checkEligibilityUrl, encoded}
-import uk.gov.hmrc.helptosavefrontend.config.{FrontendAppConfig, FrontendAuthConnector}
-import uk.gov.hmrc.helptosavefrontend.models.HtsAuth.AuthWithConfidence
-import uk.gov.hmrc.play.http.HeaderCarrier
+import uk.gov.hmrc.helptosavefrontend.models.HtsAuth.AuthWithCL200
 
-import scala.concurrent.Future
 import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
 
-class HelpToSaveAuthSpec extends TestSupport {
-
-  private val mockAuthConnector = mock[PlayAuthConnector]
-  val frontendAuthConnector = stub[FrontendAuthConnector]
+class HelpToSaveAuthSpec extends AuthSupport {
 
-  def mockAuthResultWith(ex: Throwable): Unit =
-    (mockAuthConnector.authorise(_: Predicate, _: Retrieval[Unit])(_: HeaderCarrier))
-      .expects(AuthProviders(GovernmentGateway), *, *)
-      .returning(Future.failed(ex))
+  implicit val timeout = Timeout(5, SECONDS)
 
-  def mockAuthWithRetrievalsWith(predicate: Predicate)(ex: Throwable): Unit =
-    (mockAuthConnector.authorise(_: Predicate, _: Retrieval[Enrolments])(_: HeaderCarrier))
-      .expects(predicate, *, *)
-      .returning(Future.failed(ex))
-
-  val htsAuth = new HelpToSaveAuth(fakeApplication, frontendAuthConnector) {
-    override def authConnector: AuthConnector = mockAuthConnector
-  }
+  val htsAuth = new HelpToSaveAuth(fakeApplication, mockAuthConnector)
 
   private def actionWithNoEnrols = htsAuth.authorisedForHts { implicit request ⇒ implicit htsContext ⇒
     Future.successful(Ok(""))
   }(FrontendAppConfig.checkEligibilityUrl)
 
   private def actionWithEnrols = htsAuth.authorisedForHtsWithInfo { implicit request ⇒ implicit htsContext ⇒
-    Future.successful(Ok(""))
+    htsContext.userDetails match {
+      case Some(info) ⇒ info match {
+        case Left(missingUserInfo) ⇒ Future.successful(Ok(Json.toJson(missingUserInfo)))
+        case Right(userInfo)       ⇒ Future.successful(Ok(Json.toJson(userInfo)))
+      }
+      case None ⇒ Future.successful(Ok(""))
+    }
   }(FrontendAppConfig.checkEligibilityUrl)
 
   private def mockAuthWith(error: String) =
-    mockAuthWithRetrievalsWith(AuthWithConfidence)(fromString(error))
+    mockAuthWithRetrievalsWithFail(AuthWithCL200)(fromString(error))
 
   "HelpToSaveAuth" should {
 
+    "return UserInfo after successful authentication" in {
+
+      val userInfo =
+        s"""{"forename":"Tyrion","surname":"Lannister","nino":"WM123456C","dateOfBirth":"1970-01-01","email":"tyrion_lannister@gmail.com","address":{"lines":["Casterly Rock","The Westerlands","Westeros"],"postcode":"BA148FY","country":"GB"}}""".stripMargin
```

Comment:
do we want to derive this from the info in `AuthSupport`? Otherwise this test will fail if we ever change the info in `AuthSupport`